subjects/ascii-art/justify/audit.md

```diff
@@ -12,51 +12,51 @@ Example : go run . something standard --align=right
 
 ###### Does it display the same result as above?
 
-##### Try passing as arguments `"left standard --align=right"`
+##### Try passing as arguments `left standard --align=right`
 
 ###### Does it display the correct result at the right side?
 
-##### Try passing as arguments `"right standard --align=left"`
+##### Try passing as arguments `right standard --align=left`
 
 ###### Does it display the correct result at the left side?
 
-##### Try passing as arguments `"hello shadow --align=center"`
+##### Try passing as arguments `hello shadow --align=center`
 
 ###### Does it display the correct result at the center?
 
-##### Try passing as arguments `""1 Two 4" shadow --align=justify"`
+##### Try passing as arguments `"1 Two 4" shadow --align=justify`
 
 ###### Does it display the correct result justified?
 
-##### Try passing as arguments `"23/32 standard --align=right"`
+##### Try passing as arguments `23/32 standard --align=right`
 
 ###### Does it display the correct result at the right side?
 
-##### Try passing as arguments `"ABCabc123 thinkertoy --align=right"`
+##### Try passing as arguments `ABCabc123 thinkertoy --align=right`
 
 ###### Does it display the correct result at the right side?
 
-##### Try passing as arguments `"#$%&" thinkertoy --align=center"`
+##### Try passing as arguments `#$%&\" thinkertoy --align=center`
 
 ###### Does it display the correct result at the center?
 
-##### Try passing as arguments `""23Hello World!" standard --align=left"`
+##### Try passing as arguments `"23Hello World!" standard --align=left`
 
 ###### Does it display the correct result at the left side?
 
-##### Try passing as arguments `""HELLO there HOW are YOU?!" thinkertoy --align=justify"`
+##### Try passing as arguments `"HELLO there HOW are YOU?!" thinkertoy --align=justify`
```

Comment:
Exclamation marks need to be escaped.

Reply:
I didn't know the `!` was a special character for go!

Reply:
It isn't special for go, it's rather that it is special for command-line arguments. It can be strange even depending on whether it is wrapped in single or double quotes. You can see with `echo`:
```console
bash-3.2$ echo hello!!!!!
bash: !: event not found
```

```console
bash-3.2$ echo "hello!!!"
bash: !": event not found
```

```console
bash-3.2$ echo 'hello!!!!'
hello!!!!
```